internal/pkg/poster/poster.go

```diff
@@ -70,154 +80,159 @@ func (p *Poster) StopPoSting() {
-//		log.Error("error getting faults: ", err)
-//		return
-//	}
-//
-//	challengeSeed, err := p.getChallengeSeed(ctx, head, provingPeriodStart, p.minerAddr)
-//	if err != nil {
-//		log.Error("error getting challenge seed: ", err)
-//		return
-//	}
-//
-//	faultsPrime := make([]abi.SectorNumber, len(faults))
-//	for idx := range faultsPrime {
-//		faultsPrime[idx] = abi.SectorNumber(faults[idx])
-//	}
-//
-//	minerID, err := address.IDFromAddress(p.minerAddr)
-//	if err != nil {
-//		log.Error("error mapping from miner address to ID: ", err)
-//		return
-//	}
-//
-//	output, err := p.mgr.GenerateFallbackPoSt(ctx, abi.ActorID(minerID), sortedSectorInfo, abi.PoStRandomness(challengeSeed[:]), faultsPrime)
-//	if err != nil {
-//		log.Error("error generating fallback PoSt: ", err)
-//		return
-//	}
-//
-//	poStCandidates := make([]abi.PoStCandidate, len(output.PoStInputs))
-//	for i := range output.PoStInputs {
-//		poStCandidates[i] = output.PoStInputs[i].Candidate
-//	}
-//
-//	err = p.sendPoSt(ctx, stateView, poStCandidates, output.Proof)
-//	if err != nil {
-//		log.Error("error sending fallback PoSt: ", err)
-//		return
-//	}
-//}
-//
-//func (p *Poster) sendPoSt(ctx context.Context, stateView *appstate.View, candidates []abi.PoStCandidate, proofs []abi.PoStProof) error {
-//
-//	windowedPost := &abi.OnChainPoStVerifyInfo{
-//		Candidates: candidates,
-//		Proofs:     proofs,
-//	}
-//
-//	_, workerAddr, err := stateView.MinerControlAddresses(ctx, p.minerAddr)
-//	if err != nil {
-//		return err
-//	}
-//
-//	mcid, _, err := p.outbox.Send(
-//		ctx,
-//		workerAddr,
-//		p.minerAddr,
-//		types.ZeroAttoFIL,
-//		types.NewGasPrice(1),
-//		gas.NewGas(5000),
-//		true,
-//		builtin.MethodsMiner.SubmitWindowedPoSt,
-//		windowedPost,
-//	)
-//	if err != nil {
-//		return err
-//	}
-//
-//	// wait until we see the post on chain at least once
-//	err = p.waiter.Wait(ctx, mcid, func(_ *block.Block, _ *types.SignedMessage, recp *vm.MessageReceipt) error {
-//		return nil
-//	})
-//	if err != nil {
-//		return err
-//	}
-//
-//	return nil
-//}
-
-func (p *Poster) getProvingSet(ctx context.Context, stateView *appstate.View) ([]abi.SectorInfo, error) {
-	return consensus.NewPowerTableView(stateView, stateView).SortedSectorInfos(ctx, p.minerAddr)
+func (p *Poster) doPoSt(ctx context.Context, stateView *appstate.View, deadlineIndex uint64) {
+	defer p.safeCancelPoSt()
+
+	minerID, err := address.IDFromAddress(p.minerAddr)
+	if err != nil {
+		log.Errorf("Error retrieving miner ID from address %s: %s", p.minerAddr, err)
+		return
+	}
+
+	partitions, err := stateView.MinerPartitionIndicesForDeadline(ctx, p.minerAddr, deadlineIndex)
+	if err != nil {
+		log.Errorf("Error retrieving partitions for address %s at index %d: %s", p.minerAddr, deadlineIndex, err)
+		return
+	}
+
+	// if no partitions, we're done
+	if len(partitions) == 0 {
+		return
+	}
+
+	// Some day we might want to choose a subset of partitions to prove at one time. Today is not that day.
```

Comment:
sounds like an issue

Reply:
I'm not sure. The only real reason I can think to do this is parallelization of PoSts within a single deadline. Miners are going to have to get pretty big before they'll even have more than one partition, so this seems like a pretty distant concern.